src/text.jl

```diff
@@ -0,0 +1,155 @@
+module Text
+
+using ColorTypes, Colors.@colorant_str
+
+import SenseHat.LED: RGB565, led_matrix, led_clear
+
+export show_char, show_message, @colorant_str
+
+"""
+    show_char(c::Char, color::ColorTypes.AbstractRGB = colorant"white")
+
+Display a single character `c` on the `SenseHat` LED Matrix.
+"""
+function show_char(c::Char, color::ColorTypes.AbstractRGB = colorant"white")
+    if haskey(font, c)
+        tocolor(b) = b ? RGB565(color) : RGB565(colorant"black")
+        L = PermutedDimsArray(led_matrix(), (2, 1))
+        L[:] .= colorant"black"
+        L[:,2:6] .= tocolor.(font[c])
+    else
+        error("Character font for $c not available \n")
+    end
+    return
+end
+
+"""
+    show_message(s::String, speed::Real = 0.2, color::ColorTypes.AbstractRGB = colorant"white")
+
+Display a scrolling message `s` on the `SenseHat` LED Matrix. `speed` is the time in seconds per frame and `color` is the color of text.
+
+# Example
+
+```
+using SenseHat
+
+show_message("Hello, World!", 0.2, colorant"purple")
+```
+"""
+function show_message(s::String, speed::Real = 0.2, color::ColorTypes.AbstractRGB = colorant"white")
+    for c in s
+        if haskey(font, c) == false
+            error("Character font for $c not available \n")
+            return
+        end
+        img = Array{Bool}(8, 16 + 5*length(s))
+        img[1:8,1:8] = Bool.(zeros(8,8))
+        img[1:8,(9 + 5*length(s)):(16 + 5*length(s))] = Bool.(zeros(8,8))
+        for i in 1:length(s)
+            img[1:8, (4 + 5*i):(8 + 5*i)] = font[s[i]]
+        end
+        tocolor(b) = b ? color : colorant"black"
```

Comment:
Although it will give the same results, making it the same as above (`tocolor(b) = b ? RGB565(color) : RGB565(colorant"black")`) will make it type-stable.